common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java

```diff
@@ -45,35 +55,143 @@ public OverlayManager(CrashReportManager crashReportManager) {
         addCrashCallbacks();
     }
 
+    // region Initialization and Registration
+
     public void registerOverlay(
-            Overlay overlay, RenderEvent.ElementType elementType, RenderState renderAt, Feature parent) {
-        overlayInfoMap.put(overlay, new OverlayRenderInfo(elementType, renderAt));
-        overlayParent.put(overlay, parent);
+            Overlay overlay,
+            Feature parent,
+            RenderEvent.ElementType elementType,
+            RenderState renderAt,
+            boolean enabledByDefault) {
+        overlayParentMap.putIfAbsent(parent, new LinkedList<>());
+        overlayParentMap.get(parent).add(overlay);
+
+        overlayInfoMap.put(overlay, new OverlayInfoContainer(parent, elementType, renderAt, enabledByDefault));
     }
 
     public void unregisterOverlay(Overlay overlay) {
+        overlayParentMap.get(overlayInfoMap.get(overlay).parent()).remove(overlay);
+
         overlayInfoMap.remove(overlay);
-        overlayParent.remove(overlay);
         enabledOverlays.remove(overlay);
     }
 
-    public void disableOverlays(List<Overlay> overlays) {
-        enabledOverlays.removeIf(overlays::contains);
-        overlays.forEach(
+    public void disableOverlays(Feature parent) {
+        overlayParentMap.getOrDefault(parent, Collections.emptyList()).forEach(this::disableOverlay);
+    }
+
+    public void disableOverlay(Overlay disabled) {
+        enabledOverlays.remove(disabled);
+
+        enabledOverlays.forEach(
                 overlay -> overlay.getConfigOptionFromString("userEnabled").ifPresent(overlay::onConfigUpdate));
     }
 
-    public void enableOverlays(List<Overlay> overlays, boolean ignoreState) {
-        List<Overlay> enabledOverlays = ignoreState
-                ? overlays
-                : overlays.stream().filter(Overlay::isEnabled).toList();
+    public void enableOverlays(Feature parent) {
+        overlayParentMap.getOrDefault(parent, Collections.emptyList()).forEach(this::enableOverlay);
+    }
+
+    public void enableOverlay(Overlay enableOverlay) {
+        if (!enableOverlay.isEnabled()) return;
```

Comment:
This looks suspicious. Did you not mean
```suggestion
        if (enableOverlay.isEnabled()) return;
```
?

Reply:
I'm actually feeling this "how can this ever worked?" right now. Has enableOverlay ever been called, and done something useful?

Reply:
I renamed this to better reflect what it does.